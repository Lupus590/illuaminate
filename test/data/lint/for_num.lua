-- config: (lint (only control:for-num))

local t, x = ...

-- Counting up
for _ = 1, 10 do end -- ok
for _ = 1, 10, 1 do end -- ok
for _ = 1, 10, -1 do end

-- Counting nowhere
for _ = 1, 10, 0 do end

-- Counting down
for _ = 1, -10 do end
for _ = 1, -10, -1 do end -- ok

-- Counting down on tables
for _ = #t, 1, -1 do end -- ok
for _ = #t, 1, x do end -- ok
for _ = #t, 1 do end

-- No change
for _ = 1, 1 do end
for _ = 1 + 4, 5 do end
