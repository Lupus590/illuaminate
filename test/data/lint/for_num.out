for_num.lua:[8:0-8:24]: Numeric for loop counts up from 1 to 10, but has a negative step. [control:for-num]
   │
8  │ for _ = 1, 10, -1 do end
   │ ^^^^^^^^^^^^^^^^^^^^^^^^
for_num.lua:[11:0-11:23]: Numeric for loop has a step of 0. It will never progress. [control:for-num]
    │
11  │ for _ = 1, 10, 0 do end
    │ ^^^^^^^^^^^^^^^^^^^^^^^
for_num.lua:[14:0-14:21]: Numeric for loop counts down from 1 to -10, but has a non-negative step. [control:for-num]
    │
14  │ for _ = 1, -10 do end
    │ ^^^^^^^^^^^^^^^^^^^^^
for_num.lua:[20:0-20:20]: Numeric for loop counts down from #(expr) to 1, but has a non-negative step. [control:for-num]
    │
20  │ for _ = #t, 1 do end
    │ ^^^^^^^^^^^^^^^^^^^^
for_num.lua:[23:0-23:19]: This loop has the same start and stop point, and so will only execute once. [control:for-num]
    │
23  │ for _ = 1, 1 do end
    │ ^^^^^^^^^^^^^^^^^^^
for_num.lua:[24:0-24:23]: This loop has the same start and stop point, and so will only execute once. [control:for-num]
    │
24  │ for _ = 1 + 4, 5 do end
    │ ^^^^^^^^^^^^^^^^^^^^^^^
Compilation succeeded with 6 warnings
